tests/test_other.py

```diff
@@ -136,6 +138,20 @@ def do_other_test(self, dirname, emcc_args=[], run_args=[]):
     seen = run_js('a.out.js', args=run_args, stderr=PIPE, full_output=True) + '\n'
     self.assertContained(expected, seen)
 
+  def do_smart_test(self, source, name='test.cpp', literals=[], regexes=[],
+                    emcc_args=[], run_args=[], assert_returncode=0):
+    shutil.copyfile(source, name)
```

Comment:
Do you need to copy here?  Can you just compile `source` directly?

Reply:
The copy is needed in some cases where the file name is part of the test, like in stack traces. I think it adds complexity to copy conditionally.

Reply:
Actually, I think I can remove this.